tests/convert/data_color_mine.rs

```diff
@@ -0,0 +1,245 @@
+/*
+List of color from colormine.org
+*/
+use csv;
+use palette::{Lch, Lab, Xyz, Yxy, Rgb, Hsl, Hsv, IntoColor};
+
+pub const COLOR_MINE_FILE_FULL: &'static str = "tests/convert/data_color_mine.csv";
+pub const COLOR_MINE_FILE_MINI: &'static str = "tests/convert/data_color_mine_mini.csv";
+
+#[derive(RustcDecodable, PartialEq)]
+pub struct ColorMineRaw {
+    pub color: String,
+    pub hex: String,
+    pub rgbu8_r: u8,
+    pub rgbu8_g: u8,
+    pub rgbu8_b: u8,
+    pub rgb_r: f32,
+    pub rgb_g: f32,
+    pub rgb_b: f32,
+    pub cmy_c: f32,
+    pub cmy_m: f32,
+    pub cmy_y: f32,
+    pub cmyk_c: f32,
+    pub cmyk_m: f32,
+    pub cmyk_y: f32,
+    pub cmyk_k: f32,
+    pub xyz_x: f32,
+    pub xyz_y: f32,
+    pub xyz_z: f32,
+    pub lab_l: f32,
+    pub lab_a_unscaled: f32,
+    pub lab_b_unscaled: f32,
+    pub lab_a: f32,
+    pub lab_b: f32,
+    pub lch_l: f32,
+    pub lch_c_unscaled: f32,
+    pub lch_c: f32,
+    pub lch_h: f32,
+    pub hunterlab_l: f32,
+    pub hunterlab_a: f32,
+    pub hunterlab_b: f32,
+    pub yxy_luma: f32,
+    pub yxy_x: f32,
+    pub yxy_y: f32,
+    pub luv_l: f32,
+    pub luv_u: f32,
+    pub luv_v: f32,
+    pub hsl_h: f32,
+    pub hsl_s: f32,
+    pub hsl_l: f32,
+    pub hsv_h: f32,
+    pub hsv_s: f32,
+    pub hsv_v: f32,
+}
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub struct ColorMine {
+    lab: Lab<f32>,
+    xyz: Xyz<f32>,
+    yxy: Yxy<f32>,
+    lch: Lch<f32>,
+    rgb: Rgb<f32>,
+    linear_rgb: Rgb<f32>,
+    hsl: Hsl<f32>,
+    hsv: Hsv<f32>,
+}
+
+fn gamma_to_linear(v: f32) -> f32 {
+    let eps = 0.04045_f32;
+    let a = 0.055_f32;
+    if v > eps {
+        ((v+a)/(1.0+a)).powf(2.4)
+    } else {
+        v / 12.92
+    }
+}
+
+// fn linear_to_gamma(v: f32) -> f32 {
+//     let eps = 0.0031308_f32;
+//     let a = 0.055_f32;
+//     if v > eps {
+//         (1.0+a)*v.powf(1.0/2.4) - a
+//     } else {
+//         v * 12.92
+//     }
+// }
+
+impl From<ColorMineRaw> for ColorMine {
+    fn from(src: ColorMineRaw) -> ColorMine {
+        ColorMine {
+            xyz: Xyz::new(src.xyz_x, src.xyz_y, src.xyz_z),
+            yxy: Yxy::new(src.yxy_x, src.yxy_y, src.yxy_luma),
+            lab: Lab::new(src.lab_l, src.lab_a, src.lab_b),
+            lch: Lch::new(src.lch_l, src.lch_c, src.lch_h.into()),
+            rgb: Rgb::new(src.rgb_r, src.rgb_g, src.rgb_b),
+            linear_rgb: Rgb::new(gamma_to_linear(src.rgb_r), gamma_to_linear(src.rgb_g), gamma_to_linear(src.rgb_b)),
```

Comment:
This could be `Srgb::new(src.rgb_r, srg.rgb_g, src.rgb_b).into()`, if I'm not completely off.